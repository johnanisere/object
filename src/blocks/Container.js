import styled from "styled-components";

const Container = styled.section`
  display: flex;
  padding: 20px 10px;
  @media (max-width: 768px) {
    flex-direction: column;
  }
`;
export default Container;
