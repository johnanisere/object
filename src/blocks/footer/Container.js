import styled from "styled-components";

const Container = styled.div`
  -webkit-tap-highlight-color: transparent;
  box-sizing: border-box;
  text-align: ${({ align }) => align || "left"};
  flex: ${({ flex }) => (flex ? flex : 1)};
`;

export default Container;
