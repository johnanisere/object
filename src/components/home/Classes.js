import React from "react"
import OptionLayout from "../SectionLayout"
import List from "../../blocks/FlexibleContainer"

const App = () => <OptionLayout title="CLASSES">
    <List list="true" />
</OptionLayout>

export default App