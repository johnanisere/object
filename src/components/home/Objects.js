import React from "react"
import PropTypes from "prop-types"
import { MdRemoveRedEye } from "react-icons/md";
import OptionLayout from "../SectionLayout"
import List from "../../blocks/FlexibleContainer"

const App = ({ displayData }) => {
    const title = <span><MdRemoveRedEye color="#8e8e93" /> {`  OBJECTS [${displayData.length}]`}</span>

    return (
        <OptionLayout title={title}>
            <List list="true" direction="column">
                {
                    displayData.map(({ number }, key) => <div key={key} className="list"><MdRemoveRedEye color="#8e8e93" />  OBJ {number}</div>)
                }
            </List>
        </OptionLayout>
    )
}
App.propTypes = {
    displayData: PropTypes.array.isRequired,
}
export default App