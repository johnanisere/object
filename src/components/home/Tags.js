import React from "react"
import OptionLayout from "../SectionLayout"
import List from "../../blocks/FlexibleContainer"

const App = () => (
    <OptionLayout title="TAGS">
        <List list="true" />
    </OptionLayout>)

export default App