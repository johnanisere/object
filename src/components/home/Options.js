import React from "react"
import PropTypes from "prop-types"
import Container from "../../blocks/FlexibleContainer"
import Objects from "./Objects"
import Classes from "./Classes"
import Tags from "./Tags"

const App = ({ displayData }) => {
    return (
        <Container background={true} height="100%" flex={1} direction="column">
            <Objects displayData={displayData} />
            <Classes />
            <Tags />
        </Container>
    )
}

App.propTypes = {
    displayData: PropTypes.array.isRequired,
}
export default App