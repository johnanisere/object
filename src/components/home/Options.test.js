
import React from "react";
import { shallow } from "enzyme";

import Options from "./Options"
import Container from "../../blocks/FlexibleContainer";
import Objects from "./Objects"
import Classes from "./Classes"
import Tags from "./Tags"


const props = {
    displayData: [],
}

it("Options component exists", () => {
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper.exists()).toBe(true);
});

it("Options component renders", () => {
    const wrapper = shallow(<Options {...props} />);
    expect(wrapper).toMatchSnapshot();
});

it(`Options should contain "Container",  "Options", `, () => {
    const wrapper = shallow(<Options {...props} />);

    expect(wrapper.find(Classes)).toHaveLength(1);
    expect(wrapper.find(Objects)).toHaveLength(1);
    expect(wrapper.find(Tags)).toHaveLength(1);
    expect(wrapper.find(Container)).toHaveLength(1);
});