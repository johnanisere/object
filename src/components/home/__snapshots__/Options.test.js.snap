// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options component renders 1`] = `ShallowWrapper {}`;
